ms-oidc/src/main/java/net/trajano/ms/oidc/internal/OpenIdConnectResource.java

```diff
@@ -90,16 +102,31 @@ public URI authUri(@QueryParam("state") final String state,
         if (issuerConfig == null) {
             throw new BadRequestException("Invalid issuer_id");
         }
+
+        if (!issuerConfig.isOpAllowed(op)) {
+            throw OAuthTokenResponse.badRequest(ErrorCodes.INVALID_REQUEST, "Operation not allowed");
+        }
+
+        // TODO validate whether the client credentials are valid.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)